pilot/pkg/bootstrap/server.go

```diff
@@ -1281,3 +1288,73 @@ func (s *Server) initStatusManager(_ *PilotArgs) {
 		return nil
 	})
 }
+
+// initLocalSecretListener creates a new secret listener to listen for
+// secrets which represent new multi cluster participating clusters in a mesh.
+func (s *Server) initLocalSecretListener(args *PilotArgs) {
+	// Check if this is an instance of Istiod running as external istiod
+	// The logic is based on the following:
+	// 1. command istioctl secret create will always create a secret named
+	//    istio-kubeconfig when secret type is config, to indicate that a cluster
+	//    gets used as a config cluster, the secret type must be config
+	// 2. this secret is required to be added to the namespace where istiod will be
+	//    running, otherwise, istiod wont run as an external istiod
+	// 3. this secret has to be mounted to a location to which the env KUBECONFIG points
+	// 4. due to the structure of the secret, this secret will be always mounted
+	//    as a file named config such as /var/run/secrets/remote/config
+	// 5. this kubeconfig is different from loading credentials off of a service account
+	//    which is how non-external istiod gets configured normally.
+	// Based on the above, this method will just check if the file which the value of
+	// KUBECONFIG points exists and if features.ExternalIstiod is true. With these two
```

Comment:
features.ExternalIstiod is enabled by default. It controls whether an istiod should try to get the leader lock for remote clusters that it knows about. So it doesn't really mean it's an "external control plane", it could be a primary cluster, in which case it only needs to listen for secrets inCluster.